src/sidebar/services/thumbnail.ts

```diff
@@ -49,6 +77,21 @@ export class ThumbnailService {
    * thumbnail.
    */
   get(tag: string): ImageBitmap | null {
-    return this.#cache.get(tag) ?? null;
+    const bitmap = this.#cache.get(tag);
+    if (bitmap) {
+      // Move this entry to the back of the cache's key list, so it becomes
+      // the most recently used.
+      this.#cache.delete(tag);
+      this.#cache.set(tag, bitmap);
+    }
+    return bitmap ?? null;
```

Comment:
Perhaps this condition could be flipped for simplicity.

```suggestion
    if (!bitmap) {
      return null;
    }

    // Move this entry to the back of the cache's key list, so it becomes
    // the most recently used.
    this.#cache.delete(tag);
    this.#cache.set(tag, bitmap);

    return bitmap;
```